osgp/platform/osgp-secret-management/src/main/java/org/opensmartgridplatform/secretmanagement/application/endpoints/SecretManagementEndpoint.java

```diff
@@ -37,86 +44,124 @@
+            @RequestPayload final GenerateAndStoreSecretsRequest request) throws OsgpException {
+        log.info("Handling incoming SOAP request 'storeSecretsRequest' for device {}", request.getDeviceId());
+        log.trace(request.toString());
+        final GenerateAndStoreSecretsResponse response = new GenerateAndStoreSecretsResponse();
+        final SecretTypes soapSecretTypes = request.getSecretTypes();
+        if (soapSecretTypes == null) {
+            throw new TechnicalException("Missing input: typed secrets");
+        }
+        final List<SecretType> secretTypeList = this.converter.convertToSecretTypes(soapSecretTypes);
+        final List<TypedSecret> typedSecretList = secretTypeList.stream().map(
+                t -> this.secretManagementService.generateAes128BitsSecret(t)).collect(Collectors.toList());
+        this.secretManagementService.storeSecrets(request.getDeviceId(), typedSecretList);
+        response.setResult(OsgpResultType.OK);
+        log.trace(response.toString());
         return response;
     }
 
-    private String getSecretsRequestToString(GetSecretsRequest request) {
-        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+    private String getSecretsRequestToString(final GetSecretsRequest request) {
+        final ByteArrayOutputStream baos = new ByteArrayOutputStream();
         try {
-            JAXBContext ctx = JAXBContext.newInstance(GetSecretsRequest.class);
-            Marshaller marshaller = ctx.createMarshaller();
+            final JAXBContext ctx = JAXBContext.newInstance(GetSecretsRequest.class);
+            final Marshaller marshaller = ctx.createMarshaller();
             marshaller.setProperty(Marshaller.JAXB_FORMATTED_OUTPUT, Boolean.TRUE);
             marshaller.marshal(request, baos);
-        } catch (JAXBException e) {
+        } catch (final JAXBException e) {
             log.error("Could not serialize GetSecretsRequest", e);
         }
         return baos.toString();
     }
+
+    @PayloadRoot(namespace = NAMESPACE_URI, localPart = "activateSecretsRequest")
+    @ResponsePayload
+    public ActivateSecretsResponse activateSecretsRequest(@RequestPayload final ActivateSecretsRequest request)
+            throws OsgpException {
+        log.info("Handling incoming SOAP request 'activateSecretRequest' for device {}", request.getDeviceId());
+        log.trace(request.toString());
+        final ActivateSecretsResponse response = new ActivateSecretsResponse();
+        final SecretTypes soapSecretTypes = request.getSecretTypes();
+        if (soapSecretTypes == null) {
+            throw new TechnicalException("Missing input: typed secrets");
+        }
+        try {
+            this.secretManagementService.activateNewSecrets(request.getDeviceId(),
+                    this.converter.convertToSecretTypes(soapSecretTypes));
+            response.setResult(OsgpResultType.OK);
+        } catch (final RuntimeException rte) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)